### PR TITLE
(2.16) Upgrade to graphql-java 19.4

### PR DESCRIPTION
**Pull request by jmartisk**
fixes https://access.redhat.com/security/cve/CVE-2023-28867

### CODE REVIEW

**Review by gsmet — APPROVED** [review]
Failures are unrelated.

File: bom/application/pom.xml [hunk 1]
```diff
@@ -33,6 +33,7 @@
         <micrometer.version>1.10.2</micrometer.version><!-- keep in sync with hdrhistogram -->
         <hdrhistogram.version>2.1.12</hdrhistogram.version><!-- keep in sync with micrometer -->
         <google-auth.version>0.22.0</google-auth.version>
+        <graphql-java.version>19.4</graphql-java.version>
```
Comment (gsmet) [inline bom/application/pom.xml:36]: Hmmm. I'm not sure it's a good idea to do that. Didn't you release a SmallRye GraphQL version with the updated dependency?
If we have to do that, please make sure the version is put just below `smallrye-graphql.version` with a comment saying they need to be kept in sync. But I would rather avoid it if we can.

**Review by gsmet — CHANGES_REQUESTED** [review]

**Review by gsmet — APPROVED** [review]
I will merge this as I want the next 2.16 out soon.

File: bom/application/pom.xml (hunk 1 above)
Comment (jmartisk) [reply bom/application/pom.xml:36]: @gsmet The reason for this is that the Vert.x graphql extension also depends on graphql-java, but still on 19.2, so we need to do this to satisfy the enforcer plugin. I think it's generally a good idea to manually manage the version of a library that is pulled in by multiple extensions.

**Pull request merged**